platform/src/main/java/org/stellar/anchor/platform/custody/fireblocks/FireblocksEventService.java

```diff
@@ -1,42 +1,87 @@
 package org.stellar.anchor.platform.custody.fireblocks;
 
+import static java.util.stream.Collectors.toSet;
 import static org.stellar.anchor.platform.utils.RSAUtil.SHA512_WITH_RSA_ALGORITHM;
 import static org.stellar.anchor.platform.utils.RSAUtil.isValidSignature;
 import static org.stellar.anchor.util.Log.debugF;
-import static org.stellar.anchor.util.Log.info;
+import static org.stellar.anchor.util.Log.error;
+import static org.stellar.anchor.util.Log.errorEx;
+import static org.stellar.anchor.util.Log.warnEx;
+import static org.stellar.anchor.util.Log.warnF;
 import static org.stellar.anchor.util.StringHelper.isEmpty;
 
+import java.io.IOException;
 import java.security.InvalidKeyException;
 import java.security.NoSuchAlgorithmException;
 import java.security.PublicKey;
 import java.security.SignatureException;
+import java.time.Instant;
 import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
 import org.stellar.anchor.api.custody.fireblocks.FireblocksEventObject;
+import org.stellar.anchor.api.custody.fireblocks.TransactionDetails;
+import org.stellar.anchor.api.custody.fireblocks.TransactionStatus;
+import org.stellar.anchor.api.exception.AnchorException;
 import org.stellar.anchor.api.exception.BadRequestException;
 import org.stellar.anchor.api.exception.InvalidConfigException;
+import org.stellar.anchor.api.exception.SepException;
+import org.stellar.anchor.horizon.Horizon;
 import org.stellar.anchor.platform.config.FireblocksConfig;
+import org.stellar.anchor.platform.custody.CustodyEventService;
+import org.stellar.anchor.platform.custody.CustodyPayment;
+import org.stellar.anchor.platform.custody.CustodyPayment.CustodyPaymentStatus;
+import org.stellar.anchor.platform.custody.Sep24CustodyPaymentHandler;
+import org.stellar.anchor.platform.custody.Sep31CustodyPaymentHandler;
+import org.stellar.anchor.platform.data.JdbcCustodyTransactionRepo;
 import org.stellar.anchor.util.GsonUtils;
+import org.stellar.sdk.responses.operations.OperationResponse;
+import org.stellar.sdk.responses.operations.PathPaymentBaseOperationResponse;
+import org.stellar.sdk.responses.operations.PaymentOperationResponse;
 
-public class FireblocksEventService {
+public class FireblocksEventService extends CustodyEventService {
 
   public static final String FIREBLOCKS_SIGNATURE_HEADER = "fireblocks-signature";
+  private static final Set<String> PAYMENT_TRANSACTION_OPERATION_TYPES =
+      Set.of("payment", "path_payment");
+  private static final Set<TransactionStatus> SUCCESS_WEBHOOK_EVENT_STATUSES =
+      Set.of(TransactionStatus.COMPLETED);
+  private static final Set<TransactionStatus> ERROR_WEBHOOK_EVENT_STATUSES =
+      Set.of(
```

Comment:
According to https://developers.fireblocks.com/docs/monitor-tx-status

`while (tx.status !== TransactionStatus.COMPLETED) {
        await new Promise(resolve => setTimeout(resolve, 3000));
        tx = await fireblocks.getTransactionById(txId);
        if (tx.status === TransactionStatus.BLOCKED || tx.status === TransactionStatus.CANCELLED || tx.status === TransactionStatus.FAILED) {
            throw new Error("Signing request failed.");
        }
        console.log('TX ' + tx.id + ' is currently at status - '+ tx.status);
    }`

Only FAILED, CANCELLED AND BLOCKED are handled as error statuses.
Maybe this documentation isn't up-to-date.
Lets keep it as is for now